### PR TITLE
feat: add generic list services endpoint

### PR DESCRIPTION
**Pull request by zcstarr**

### CODE REVIEW

File: openrpc.json [hunk 1]
```diff
@@ -29,6 +29,55 @@
...
+							"installed"
+						]
+					},
+					"required": true
+				}
+			],
+			"result": {
+				"name": "listedServices",
+				"schema": {
+					"type": "array",
```
Comment (shanejonas) [inline openrpc.json]: should add 

```
"title": "listedServices"
```

so that the generated types get better names

**Comment by zcstarr** [comment]
This adds the generic listServices endpoint, you can specify a single string filter with "all" or "available" or "running" or "installed" , and it will return back the services. If you want to know about combinations, that's left up to the client to filter the all callback.

I didn't want to go to wild on this, because I know this will probably change a bit once we start supporting dynamic configuration. 

In addition this kind of sneaks in additional test for the rpc endpoint, that were missing. I.e. it makes sure the methods in the method mapping functions are tested. 

This change will also deprecate listInstalledService, and listRunningServices. There's time so I just added this function before removing the old functionality to allow for time to update any dependent code.

File: openrpc.json (hunk 1 above)
Comment (zcstarr) [reply openrpc.json]: I would maybe after this commit lands, then I can go back and update all the stuff for good names, if we don't hit issues like before.

**Pull request merged**